include/rfl/internal/num_fields.hpp

```diff
@@ -97,20 +108,68 @@ struct CountFieldsHelper {
     }
   }
 
+  template <std::size_t size = 0>
+  static consteval std::size_t get_the_sole_nested_array_size() {
```

Comment:
Wrong name: this is not a "nested array" size check.

Reply:
changed to `get_the_sole_nested_base_field_count`